include/gauge_field_order.h

```diff
@@ -1375,72 +1373,62 @@ namespace quda {
         {
         }
 
-        __device__ __host__ inline void Pack(RegType out[12], const RegType in[18], int idx) const
+        __device__ __host__ inline void Pack(real out[12], const complex in[9], int idx) const
         {
           reconstruct_12.Pack(out, in, idx);
         }
 
         template <typename I>
-        __device__ __host__ inline void Unpack(RegType out[18], const RegType in[12], int idx, int dir,
-            const RegType phase, const I *X, const int *R) const
+        __device__ __host__ inline void Unpack(complex out[18], const real in[12], int idx, int dir, real phase,
```

Comment:
Should this be `complex out[9]` ?

Reply:
Thanks.  This nicely highlights one issue: that there is no array-size type checking using naked C++ pointers.  One reason why I'd like to move us to using `std::array` eventually.

Reply:
All in on `std::array` - just beyond this PR